xla/util.h

```diff
@@ -1018,6 +1018,45 @@ struct FreeDeleter {
 std::unique_ptr<void, FreeDeleter> AlignedAlloc(std::size_t alignment,
                                                 std::size_t size);
 
+// Serialized static variable initialization from the `func` output.
+// It avoid a deadlock between static variable initialization lock
+// and a lock in the `func` function.
+// Usage:
+// Instead of
+// static SomeType* p = [](){
+//     // for example we call some python code using nanobind
+//     nb::module_ numpy = nb::module_::import_("numpy");
+//     auto np_int8 = nb::object(numpy.attr("int8"));
+//     SomeType* obj = new SomeType(np_uint8);
+//     return obj;
+// };
+// We use SafeStaticInit
+// auto func = [](){
+//     // for example we call some python code using nanobind
+//     nb::module_ numpy = nb::module_::import_("numpy");
+//     auto np_int8 = nb::object(numpy.attr("int8"));
+//     SomeType* obj = new SomeType(np_uint8);
+//     return obj;
+// }
+// SomeType* p = SafeStaticInit<SomeType>(func);
+template<typename T, typename F>
+T * SafeStaticInit(F func) {
```

Comment:
I'd be tempted to return a reference rather than a pointer. No strong reason, but it would read better to me.